### ISSUE TITLE
Undefined variables

### ISSUE THREAD

**Issue by CMLL** [microsoft/python-language-server#1846]
## Environment data

- Language Server version: 0.5.10   
- OS and version: Ubuntu 19.04
- Python version (& distribution if applicable, e.g. Anaconda): 2.7

## Expected behaviour

Recently declared variables are recognized.

## Actual behaviour

Getting undefined variables one line after the variable is declared. Think this is happening mostly with version 0.5.10, at least I didn't noticed it before the update.

## Code Snippet / Additional lnformation

![image](https://user-images.githubusercontent.com/3720126/71191112-b6b82800-227d-11ea-8926-ad52f700c3e0.png)

**Comment by MikhailArkhipov** [comment]
Is this still a problem?

**Comment by CMLL** [comment]
Yes, but after observing the problem more, I believe all my issues stem from the same issue. 
https://github.com/microsoft/python-language-server/issues/1755
https://github.com/microsoft/python-language-server/issues/1502

I have some sort of construct in the code I work on, I believe is a import loop of some kind, that makes the LS go extremely high on virtual memory, prevents it from closing correctly when the client is closed. When this state is reached, the analysis message gets stuck in there and the undefined variables problems arise when trying to use this client.

I think all my issues could be condensed into one.
